pkg/runtime/reconciler.go

```diff
@@ -218,7 +218,13 @@ func (r *resourceReconciler) Sync(
 			return latest, err
 		}
 	}
-	return r.handleRequeues(ctx, latest)
+	// Attempt to late initialize the resource. If there are no fields to
+	// late initialize, this operation will be a no-op.
+	var lateInitializedLatest acktypes.AWSResource
+	if lateInitializedLatest, err = r.lateInitializeResource(ctx, rm, desired, latest); err != nil {
+		return lateInitializedLatest, err
+	}
+	return r.handleRequeues(ctx, lateInitializedLatest)
```

Comment:
we can re-use the same `latest` variable:

```go
if latest, err = r.lateInitializeResource(ctx, rm, desired, latest); err != nil {
    return latest, err
}
return r.handleRequeues(ctx, latest)
```

like is done above on line 217.